src/main/scala/services/AuthService.scala

```diff
@@ -0,0 +1,88 @@
+package io.blindnet.identity
+package services
+
+import endpoints.objects.*
+import errors.*
+import mail.*
+import models.Account
+import util.Resources
+
+import cats.data.OptionT
+import cats.effect.*
+import cats.effect.std.UUIDGen
+
+import java.util.UUID
+import scala.util.Random
+
+class AuthService(repos: Repositories) {
+  given UUIDGen[IO] = UUIDGen.fromSync
+
+  private def generateStaticToken(): IO[String] =
+    IO(Random.alphanumeric.take(128).mkString)
+
+  private def sendVerificationEmail(email: String, emailToken: String): IO[Unit] =
+    def processTemplate(template: String, env: Env): String =
+      template.replace("$verification_link$", s"${env.baseUrl}/verify?token=$emailToken")
+
+    for {
+      env <- Env.get
```

Comment:
I prefer reading the config once and passing the object as a parameter. Makes it easier to debug/test